lib/Source.php

```diff
@@ -9,6 +9,20 @@
  */
 class Source extends ApiResource
 {
+    /**
+     * @return Util\Set Attributes that are nested but still updatable from
+     *    the parent class's URL (e.g. metadata).
+     */
+    public static function getNestedUpdatableAttributes()
+    {
+        static $nestedUpdatableAttributes = null;
+        if ($nestedUpdatableAttributes === null) {
+            $nestedUpdatableAttributes = StripeObject::getNestedUpdatableAttributes();
+            $nestedUpdatableAttributes->add('card');
```

Comment:
I worry that this will make it even harder to not miss edge-case. We already "break" the API often by forgetting to add those new API params to the whitelist. Now we need to add those but to potentially multiple objects now instead of just one canonical (though hidden) place in the library

Reply:
What this PR does is give the option to override the global whitelist in each class. If we're afraid that splitting up the whitelist will make it harder to maintain, we can simply keep the PR in its current form, i.e. keep the global whitelist on `StripeObject` and keep the one special case of `card` under `Source`.

Reply:
I understand but it _feels_ wrong/confusing.